moz-webgpu-cts/src/main.rs

```diff
@@ -398,18 +398,30 @@ fn run(cli: Cli) -> ExitCode {
                         .map(BufReader::new)
                         .map_err(Report::msg)
                         .wrap_err("failed to open file")
-                        .and_then(|reader| {
-                            serde_json::from_reader::<_, ExecutionReport>(reader)
-                                .map(Some)
-                                .or_else(|e| {
-                                    if e.is_eof() && matches!((e.line(), e.column()), (1, 0)) {
-                                        Ok(None)
-                                    } else {
-                                        Err(e)
-                                    }
-                                })
+                        .and_then(|mut reader| {
+                            // Servo has multiple reports in same json file (each in one line)
+                            // First one is from first run, second one run that runs only unexpected results for filtering
+                            //
+                            // TODO(sagudev): merge reports instead of using only first one
+                            let mut first_report = String::new();
+                            reader
+                                .read_line(&mut first_report)
```

Comment:
suggestion: This looks correct to me, based on information you've provided. However, I suspect you'll have more robust parsing and better diagnostic information WRT byte offset if you use [`serde_json::StreamDeserializer`](https://docs.rs/serde_json/1.0.121/serde_json/struct.StreamDeserializer.html).

Reply:
Diagnostic should still be ok, because it's the first line, but solution with [StreamDeserializer](https://docs.rs/serde_json/1.0.121/serde_json/struct.StreamDeserializer.html) is more future proof and I love it (to be honest I didn't know that StreamDeserializer existed).